unstructured/ingest/connector/github.py

```diff
@@ -52,23 +54,71 @@ class GitHubIngestDoc(GitIngestDoc):
     connector_config: SimpleGitHubConfig
     registry_name: str = "github"
 
-    def _fetch_and_write(self) -> None:
-        content_file = self.connector_config.get_repo().get_contents(self.path)
+    @property
+    def date_created(self) -> t.Optional[str]:
+        return None
+
+    @requires_dependencies(["github"], extras="github")
+    def _fetch_content(self, is_content_file=False):
+        from github.GithubException import UnknownObjectException
+
+        try:
+            content_file = self.connector_config.get_repo().get_contents(self.path)
+        except UnknownObjectException:
+            logger.error(f"File doesn't exists {self.connector_config.url}/{self.path}")
+            return None
+        except Exception:
+            logger.error(f"Error processing {self.connector_config.url}/{self.path}")
+            raise
+
+        if is_content_file:
+            return content_file
+
         contents = b""
         if (
             not content_file.content  # type: ignore
             and content_file.encoding == "none"  # type: ignore
             and content_file.size  # type: ignore
         ):
             logger.info("File too large for the GitHub API, using direct download link instead.")
+            # NOTE: Maybe add a raise_for_status to catch connection timeout or HTTP Errors?
             response = requests.get(content_file.download_url)  # type: ignore
             if response.status_code != 200:
                 logger.info("Direct download link has failed... Skipping this file.")
+                return None
             else:
                 contents = response.content
         else:
             contents = content_file.decoded_content  # type: ignore
+        return contents
 
+    def update_source_metadata(self, **kwargs):
+        content_file = self._fetch_content(True)
+        if content_file is None:
+            self.source_metadata = SourceMetadata(
+                exists=False,
+            )
+            return
+
+        date_modified = datetime.strptime(
+            content_file.last_modified,
+            "%a, %d %b %Y %H:%M:%S %Z",
+        ).isoformat()
+        self.source_metadata = SourceMetadata(
+            date_modified=date_modified,
+            version=content_file.etag,
+            source_url=content_file.download_url,
+            exists=True,
+        )
+
+    def _fetch_and_write(self) -> None:
+        contents = self._fetch_content()
+        self.update_source_metadata()
```

Comment:
can you do the pattern here like you do in other connectors? i.e. pass contents into update_source_metadata so it doesn't fetch twice?